src/hotspot/cpu/riscv/c1_LIRAssembler_riscv.cpp

```diff
@@ -1322,7 +1322,12 @@ void LIR_Assembler::comp_fl2i(LIR_Code code, LIR_Opr left, LIR_Opr right, LIR_Op
   }
 }
 
-void LIR_Assembler::align_call(LIR_Code code) {  }
+void LIR_Assembler::align_call(LIR_Code code) {
+  // With RVC a call may get 2-byte aligned.
+  //   the address of jal itself (which will be patched later) should not span the cache line.
+  //   See CallStaticJavaDirectNode::compute_padding() for more info.
+  __ align(4);
```

Comment:
It's better to add some alignment asserts before the code that may be patched, for example, in MacroAssembler::emit_static_call_stub.